### PR TITLE
Remove start script

### PR DESCRIPTION
**Pull request by komninoschat**
With this PR, the `start.sh` script is removed and replaced by multiple NPM commands:
* `test` - Runs the tests inside a Docker container.
* `test:watch` - Runs the tests inside a Docker container and re-runs then whenever the source code is changed.
* `dev` - Runs the dev server inside a Docker container and restarts it whenever the source code is changed.

### CODE REVIEW

**Comment by komninoschat** [comment]
@elitan Should I include the documentation for this change on our `README.md` or on the documentation?

**Comment by elitan** [comment]
@komninoschat Maybe we could add this information in a `contributing.md` file for other contributors?

**Comment by komninoschat** [comment]
@elitan Good idea! Created a new issue for that(#483) and will merge this one as soon as you approve.

File: package.json [hunk 1]
```diff
@@ -110,6 +116,9 @@
     "@vuepress/plugin-back-to-top": "1.4.1",
     "codecov": "^3.7.1",
     "cz-conventional-changelog": "3.1.0",
+    "death": "^1.1.0",
+    "docker-compose": "^0.23.6",
+    "dotenv-cli": "^4.0.0",
```
Comment (elitan) [inline package.json:121]: Are these packages used?

I think the `docker-compose` you run ex during `yarn dev` is the `docker-compose` being installed locally on the host machine. I might be wrong though.

Comment (komninoschat) [reply package.json]: The `death` and `docker-compose` ones aren't used - they were leftovers from previous experimentation. Nice catch!

**Review by elitan — APPROVED** [review]
LGTM!

**Pull request merged**